src/nidigital/examples/nidigital_ppmu_source_and_measure.py

```diff
@@ -0,0 +1,187 @@
+    parser.add_argument('-c', '--channels', default='DUTPin1, SystemPin1', help='Channel(s)/Pin(s) to use')
+    parser.add_argument('-m', '--measure', default='voltage', choices=['voltage', 'current'], help='Measure voltage or measure current')
+    parser.add_argument('-at', '--aperture-time', default=0.000004, type=float, help='Aperture time in seconds')
+    subparser = parser.add_subparsers(dest='command', help='Sub-command help, by default it source voltage')
+
+    source_current = subparser.add_parser('source-current', help='Source current')
+    source_current.add_argument('-clr', '--current-level-range', default=0.000002, type=float, help='Current level range in amps')
+    source_current.add_argument('-cl', '--current-level', default=0.000002, type=float, help='Current level in amps')
+    source_current.add_argument('-vlh', '--voltage-limit-high', default=3.3, type=float, help='Voltage limit high in volts')
+    source_current.add_argument('-vll', '--voltage-limit-low', default=0, type=float, help='Voltage limit low in volts')
+    source_current.add_argument('-st', '--settling-time', default=0.01, type=float, help='Settling time in seconds')
+
+    source_voltage = subparser.add_parser('source-voltage', help='Source current')
+    source_voltage.add_argument('-clr', '--current-level-range', default=0.000002, type=float, help='Current level range in amps')
+    source_voltage.add_argument('-vl', '--voltage-level', default=3.3, type=float, help='Voltage level in volts')
+    source_voltage.add_argument('-st', '--settling-time', default=0.01, type=float, help='Settling time in seconds')
+
+    args = parser.parse_args(argsv)
+
+    if args.command == 'source-current':
+        example(
+            args.resource_name,
+            'Simulate=1, DriverSetup=Model:6571' if args.simulate == 'True' else '',
+            args.channels,
+            args.measure,
+            args.aperture_time,
+            args.command,
+            args.settling_time,
+            args.current_level_range,
+            args.current_level,
+            args.voltage_limit_high,
+            args.voltage_limit_low)
+    elif args.command == 'source-voltage':
+        example(
+            args.resource_name,
+            'Simulate=1, DriverSetup=Model:6571' if args.simulate == 'True' else '',
+            args.channels,
+            args.measure,
+            args.aperture_time,
+            args.command,
+            args.settling_time,
+            args.current_level_range,
+            voltage_level=args.voltage_level)
+    else:
+        if args.measure == 'current':
+            raise ValueError('Cannot measure current on a channel that is not sourcing voltage or current')
+        example(
+            args.resource_name,
+            'Simulate=1, DriverSetup=Model:6571' if args.simulate == 'True' else '',
+            args.channels,
+            args.measure,
+            args.aperture_time,
+            args.command)
```

Comment:
This can be skipped if default value is configured.

Reply:
Done